### ISSUE TITLE
Zero inductance gives NaN or wrong results in the MNA circuit solver

### ISSUE THREAD

**Issue by gvcallen** [gvcallen/paramrf#215]
An inductance of exactly `L = 0` breaks circuits solved with the default `GlobalMNACircuitSolver`. `Inductor` and `InductorQ` return NaN, and `PiSectionCLC` returns a wrong value and a wrong gradient. This happens whether `L` is free or `Fixed(0.0)`. Some other components also give wrong gradients at a zero value. Found while fitting circuitREACH models; ParamRF 0.35.3, x64.

A value exactly on a parameter bound is a separate problem, listed at the end. It is not the main cause.

## Reproduction

```python
import jax, jax.numpy as jnp
jax.config.update("jax_enable_x64", True)
import parax as prx
import pmrf as prf
from pmrf.frequency import Frequency
from pmrf.models import (Circuit, Port, Inductor, PiSectionCLC,
                         GlobalMNACircuitSolver, GlobalScatteringCircuitSolver)

freq = Frequency(start=0.1, stop=1.0, npoints=3, unit='GHz')

def two_port(element, solver):
    p0, p1 = Port(z0=50.0), Port(z0=50.0)
    return Circuit([[(p0, 0), (element, 0)], [(element, 1), (p1, 0)]], solver=solver)

def loss_and_grad(model):
    y0 = prf.param_values(model, free_only=True, space='raw')
    f = lambda y: jnp.sum(jnp.abs(prx.unwrap(prf.update(model, y, space='raw')).s(freq)[:, 0, 0])**2)
    return float(f(y0)), {k: float(v) for k, v in jax.grad(f)(y0).items()}

for solver in (GlobalMNACircuitSolver(), GlobalScatteringCircuitSolver()):
    name = type(solver).__name__
    print(name, 'Inductor(L=0)     ', loss_and_grad(two_port(Inductor(L=prf.Unconstrained(0.0)), solver)))
    pi = PiSectionCLC(C1=prf.Fixed(1e-12), L=prf.Unconstrained(0.0), C2=prf.Fixed(1e-12))
    print(name, 'PiSectionCLC(L=0) ', loss_and_grad(two_port(pi, solver)))
```

```
GlobalMNACircuitSolver        Inductor(L=0)      (nan, {'circuit[1].L': nan})
GlobalMNACircuitSolver        PiSectionCLC(L=0)  (0.11952405883415297, {'circuit[1].L': -2.79e+17})
GlobalScatteringCircuitSolver Inductor(L=0)      (0.0, {'circuit[1].L': 0.0})
GlobalScatteringCircuitSolver PiSectionCLC(L=0)  (0.11980618930241488, {'circuit[1].L': -4.79e+07})
```

The scattering solvers (global, hierarchical, sequential) are correct in every case below. The nodal solver behaves like MNA.

### 1. `Inductor.y()` / `InductorQ.y()` are infinite at L = 0

`Model.mna` (`pmrf/models/base.py:568`) builds the stamp from `y()` whenever a model overrides it. `Inductor.y()` (`pmrf/models/components/lumped.py:409`) computes `1 / (jωL)`, which is `inf` at L = 0 (and at ω = 0 for any L). The MNA and nodal solves turn that into NaN.

- This also happens with `Inductor(L=prf.Fixed(0.0))`. The NaN then reaches the gradients of every other free parameter in the circuit.
- `InductorQ` behaves the same way.
- L = 1e-15 works.

An ideal short has a finite impedance and ABCD matrix but no admittance. `PiSectionCLC` avoids the NaN because it goes through `a2mna`, which uses an auxiliary current. `Inductor` and `InductorQ` could be stamped the same way (a z- or MNA-based stamp) instead of from `y()`.

### 2. MNA returns a regularised Y when ports are joined by zero impedance

`GlobalMNACircuitSolver` reduces the circuit to its external Y-matrix and adds `eps = 1e-12` to the diagonal (`solvers/nodal.py:81,120`). If two ports are connected through a zero-impedance path, that Y is infinite. The solver then returns a finite approximation of about 1e12 instead of failing.

In the example above, `PiSectionCLC(L=0)` gives Σ|S11|² = 0.119524 instead of 0.119806, and dL = -2.8e17 instead of -4.8e7. The wrong result gives no warning.

### 3. Zero-value gradients in other components

- `BoxSectionCLCC` (`pmrf/models/components/sections.py:227`) replaces L = 0 with `finfo.eps` inside a `jnp.where`. The value is correct, but the **gradient with respect to L is exactly 0** at L = 0; the true value is about -4.5e7 in a test circuit. The model is also NaN at f = 0 Hz for any L.
- `CapacitorQ.y()` (`lumped.py:315`) at C = 0 gives the correct value but a **NaN gradient** under MNA.
- Plain `Capacitor`, series or shunt, is correct at C = 0 in every solver.

### 4. A value exactly on a bound (separate issue)

- `Bounded(0, 10, value=0)` and `Constrained(Positive(), 0)` give a raw value of `-inf`. The gradient there is 0, so the parameter cannot move. `minimize` already rejects this on purpose (`pmrf/optimize/base.py:143`).
- `Random(Uniform(0, 10), value=0)` does not reject it. It silently moves the value to 2.2e-15.

This inconsistency between factories may be worth a decision of its own.

## Questions

- How should `Inductor` and `InductorQ` be stamped for MNA and nodal solvers so that L = 0 (and ω = 0) is exact?
- Should the MNA solver detect or report an unbounded external Y instead of regularising it silently? Or should it return a representation that stays finite for shorted ports?
- Should `BoxSectionCLCC` keep its `eps` substitution once the above is fixed? It currently removes the L gradient at zero.

**Comment by luar123** [comment]
Also ran into this issue. Additionally, setting a very small value does not always help:

`Bounded` snaps tiny values to exactly zero. `Bounded(-1, 1, value=2.8e-17,
 scale=1e-9)` round-trips through the bijector to exactly 0.0.
